testing/client.go

```diff
@@ -58,22 +58,35 @@ func NewFakeClient(scheme *runtime.Scheme, objs ...client.Object) *clientWrapper
 		reactionChain:           []Reactor{},
 	}
 	// generate names on create
-	client.AddReactor("create", "*", func(action Action) (bool, runtime.Object, error) {
+	c.AddReactor("create", "*", func(action Action) (bool, runtime.Object, error) {
 		if createAction, ok := action.(CreateAction); ok {
 			obj := createAction.GetObject()
 			if accessor, ok := obj.(metav1.ObjectMetaAccessor); ok {
 				objmeta := accessor.GetObjectMeta()
 				if objmeta.GetName() == "" && objmeta.GetGenerateName() != "" {
-					client.genCount++
+					c.genCount++
 					// mutate the existing obj
-					objmeta.SetName(fmt.Sprintf("%s%03d", objmeta.GetGenerateName(), client.genCount))
+					objmeta.SetName(fmt.Sprintf("%s%03d", objmeta.GetGenerateName(), c.genCount))
 				}
 			}
 		}
 		// never handle the action
 		return false, nil, nil
 	})
-	return client
+	return c
+}
+
+func prepareObjects(objs []client.Object) []client.Object {
+	o := make([]client.Object, len(objs))
+	for i := range objs {
+		obj := objs[i].DeepCopyObject().(client.Object)
+		// default to a non-zero creation timestamp
+		if obj.GetCreationTimestamp().Time.IsZero() {
+			obj.SetCreationTimestamp(metav1.NewTime(time.UnixMilli(1000)))
```

Comment:
**[thought]:** We default a resource's version to `999`. In my understanding, that's an intentionally odd value so that it stands out. I am wondering if defaulting the creation timestamp to `time.UnixMilli(999)` would serve a similar purpose. It would also feel consistent; once you see `999` it's like _"~~Bart~~ reconciler-runtime was here"_.

https://github.com/vmware-labs/reconciler-runtime/blob/d526d748cdbca7b3f16ac92b7199ddb27505cb68/testing/subreconciler.go#L211

Reply:
This code isn't new, just moved. While I agree with the spirit, I'd rather not risk a subtile regression

Reply:
Totally makes sense to me!